docs/design/coreclr/botr/data_contracts_for_coreclr.md

```diff
@@ -0,0 +1,430 @@
+# Data Contracts for CoreCLR, and the data contract stream
```

Comment:
> How much process memory overhead is acceptable to use for self-descriptive contract information? I'd propose a total runtime-wide target of 5KB for NativeAOT, 10KB for coreclr

+1.

The current implementation of the very minimal NativeAOT contracts is close or maybe even over this number already. It is several kBs of code, bunch of strings, and a chunk of writeable memory where the contracts gets created as the first thing during the runtime startup.

Reply:
> I would guess on the order of 500 per runtime per release

OK it sounds like we don't have a concensus on this yet :) When you are thinking 500 is that something like every struct mentioned in the spec has its own associated contract and version number? Is there something else contributing a large amount of these versioned contracts?

Reply:
> > I would guess on the order of 500 per runtime per release
> 
> OK it sounds like we don't have a concensus on this yet :) When you are thinking 500 is that something like every struct mentioned in the spec has its own associated contract and version number? Is there something else contributing a large amount of these versioned contracts?

yea I think partly it's the data structures, and partly how finegrained we want the algorithmic contracts to be.  Plus I suspect there will be a bunch of duplication because we don't have parametrized contracts or generic contracts.